src/Features/CSharp/Portable/SpellCheck/CSharpSpellCheckSpanService.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.CSharp.Classification;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.SpellCheck;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.SpellCheck
+{
+    [ExportLanguageService(typeof(ISpellCheckSpanService), LanguageNames.CSharp), Shared]
+    internal class CSharpSpellCheckSpanService : AbstractSpellCheckSpanService
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpSpellCheckSpanService()
+        {
+        }
+
+        protected override string? GetClassificationForIdentifier(SyntaxToken token)
+            => ClassificationHelpers.GetClassificationForIdentifier(token);
+
+        protected override TextSpan GetSpanForComment(SyntaxTrivia trivia)
+        {
+            switch (trivia.Kind())
+            {
+                case SyntaxKind.SingleLineCommentTrivia:
+                    return TextSpan.FromBounds(trivia.Span.Start + "//".Length, trivia.Span.End);
+
+                case SyntaxKind.MultiLineCommentTrivia:
+                    var start = trivia.Span.Start + "/*".Length;
+                    var end = trivia.Span.End;
+                    if (end >= start + 2 &&
+                        trivia.ToFullString().EndsWith("*/"))
```

Comment:
Is it worth even bothering to filter out the comment start/end markers? I would expect the spell checker will have to handle things like below, and we're not removing the asterisks etc. so if the spell checker doesn't like `/*` and `*/` then its not going to be a great spell checker, right?

```
/* hi there
 * yay
 * okay */
```

Reply:
> Is it worth even bothering to filter out the comment start/end markers?

Talking to @hjihong she asked if we could trim things when sensible.  Though i agree the multiline comment case is suspect as you are going to get all these pieces, and the spell checker will still have to tokenize these regions anyways.  

@hjihong  how important/necessary is it for us to do trimming?

Reply:
Not totally important for cases like // or */. 

My main concern is that we don't spell check XML doc tags in comments.

Reply:
ah ok.  yes, in doc comments we do not include the structure.  I'll simplify the impls of the standard comment code.